subliminal/providers/subs4series.py

```diff
@@ -0,0 +1,276 @@
+                    sanitize(name) for name in [video.series] + video.alternative_series):
+                matches.add('series')
+            # year
+            if video.original_series and self.year is None or video.year and video.year == self.year:
+                matches.add('year')
+
+        # release_group
+        if (video.release_group and self.version and
+                any(r in sanitize_release_group(self.version)
+                    for r in get_equivalent_release_groups(sanitize_release_group(video.release_group)))):
+            matches.add('release_group')
+        # resolution
+        if video.resolution and self.version and video.resolution in self.version.lower():
+            matches.add('resolution')
+        # format
+        if video.source and self.version and sanitize(video.source) in sanitize(self.version):
+            matches.add('source')
+        # other properties
+        matches |= guess_matches(video, guessit(self.version, {'type': 'episode'}), partial=True)
+
+        return matches
+
+
+class Subs4SeriesProvider(Provider):
+    """Subs4Series Provider."""
+    languages = {Language.fromalpha2(l) for l in {'el', 'en'}}
+    server_url = 'https://www.subs4series.com'
+    search_url = '/search_report.php?search=%s&searchType=1'
+    episode_link = '/tv-series/%s/season-%d/episode-%d'
+    subtitle_class = Subs4SeriesSubtitle
+
+    def __init__(self):
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
+
+    def terminate(self):
+        self.session.close()
+
+    def get_show_ids(self, title, year=None):
+        """Get the best matching show id for `series` and `year`.
+
+        First search in the result of :meth:`_get_show_suggestions`.
+
+        :param title: show title.
+        :param year: year of the show, if any.
+        :type year: int
+        :return: the show id, if found.
+        :rtype: str
+
+        """
+        title_sanitized = sanitize(title).lower()
+        show_ids = self._get_suggestions(title)
+
+        matched_show_ids = []
+        for show in show_ids:
+            show_id = None
+            show_title = sanitize(show['title'])
+            # attempt with year
+            if not show_id and year:
+                logger.debug('Getting show id with year')
+                show_id = '/'.join(show['link'].rsplit('/', 2)[1:]) if show_title == '%s %d' % (
+                    title_sanitized, year) else None
+
+            # attempt clean
+            if not show_id:
+                logger.debug('Getting show id')
+                show_id = '/'.join(show['link'].rsplit('/', 2)[1:]) if show_title == title_sanitized else None
+
+            if show_id:
+                matched_show_ids.append(show_id)
+
+        return matched_show_ids
+
+    @region.cache_on_arguments(expiration_time=SHOW_EXPIRATION_TIME, to_str=text_type)
```

Comment:
Probably you should only cache if it returned results, like here : https://github.com/Diaoul/subliminal/blob/develop/subliminal/providers/legendastv.py#L249

It was an issue in the other provider that because empty results were cached, it wasn't able to download the subtitles when they got uploaded. People had to wait a long time or clear the cache